src/nat/profiler/callbacks/langchain_callback_handler.py

```
@@ -90,6 +82,34 @@ def __repr__(self) -> str:
+        del state["_lock"]
+        del state["step_manager"]
+        return state
+
+    def __setstate__(self, state):
+        """Used for deserializing"""
+        # restore the state which was picklable
+
+        print(f"Deserializing LangchainProfilerHandler {self.__dict__}")
+
+        if (getattr(self, "_lock", None) is None):
+            setattr(self, "_lock", threading.Lock())
```

Ruff (0.14.10)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)